test_communication/test/requester_py.py

```diff
@@ -17,55 +17,52 @@
 import sys
 
 
-def requester(service_name, number_of_cycles):
-    from test_msgs.service_fixtures import get_test_srv
+def requester(service_name):
     import rclpy
-
-    service_pkg = 'test_msgs'
-    module = importlib.import_module(service_pkg + '.srv')
-    srv_mod = getattr(module, service_name)
-
+    from test_msgs.service_fixtures import get_test_srv
     rclpy.init(args=[])
 
-    node = rclpy.create_node('requester')
+    try:
+        # Import the service
+        service_pkg = 'test_msgs'
+        module = importlib.import_module(service_pkg + '.srv')
+        srv_mod = getattr(module, service_name)
 
-    srv_fixtures = get_test_srv(service_name)
+        srv_fixtures = get_test_srv(service_name)
+        service_name = 'test_service_' + service_name
```

Comment:
Everything before `create_node` can stay above `rclpy.init()` and thus be outside of the try block (except the get_test_srv call maybe)

Reply:
Looks like `get_test_srv` works fine before `rclpy.init()`
d5afa0e46ba37a74bf84e9b5c15fc5d6f6eca1db